src/System.Reflection/tests/project.json

```diff
@@ -1,13 +1,16 @@
 {
   "dependencies": {
+    "System.AppContext": "4.0.0",
     "System.Collections": "4.0.10",
     "System.Diagnostics.Debug": "4.0.10",
     "System.Globalization": "4.0.10",
     "System.IO": "4.0.10",
+    "System.IO.FileSystem": "4.0.0.0",
     "System.Linq": "4.0.0",
-    "System.Reflection": "4.0.10",
+    "System.Reflection.Emit": "4.0.0",
```

Comment:
Pragmatically, all 3 tests are actually dependent on not being run on .NET Native at the moment. The .NET Native test will just be `Assert.NotNull(ThisAssembly().Location); Assert.Empty(ThisAssembly.Location())`. However, we don't currently share enough infrastructure to express that directly here in corefx repo.